### PR TITLE
append feature flags to target triple for rt isa check

### PR DESCRIPTION
**Pull request by david-salinas**
Change-Id: I1285cf6a99b4ab534f049c820cb1bbbdb26259f9

I messed up the last PR (https://github.com/RadeonOpenCompute/hcc/pull/1094)

### CODE REVIEW

File: lib/hsa/mcwamp_hsa.cpp [hunk 1]
```diff
@@ -2714,7 +2719,9 @@ class HSADevice final : public KalmarDevice
         } co_data;
 
         status = hsa_isa_from_name(isa.c_str(), &co_data.isa_type);
-        STATUS_CHECK(status, __LINE__);
+
+	if (status != HSA_STATUS_SUCCESS)
+		return false;
```
Comment (scchan) [inline lib/hsa/mcwamp_hsa.cpp]: please fix indentation

**Review by scchan — CHANGES_REQUESTED** [review]

File: lib/hsa/mcwamp_hsa.cpp (hunk 1 above)
Comment (scchan) [reply lib/hsa/mcwamp_hsa.cpp]: Also `kernel_source` needs to be freed before returning or turn that buffer into an automatic object

File: lib/hsa/mcwamp_hsa.cpp [hunk 2]
```diff
@@ -2730,6 +2739,7 @@ class HSADevice final : public KalmarDevice
                     return HSA_STATUS_SUCCESS;
                 },
                 &co_data);
+
```
Comment (scchan) [inline lib/hsa/mcwamp_hsa.cpp]: could you remove this?

File: lib/hsa/mcwamp_hsa.cpp (hunk 1 above)
Comment (david-salinas) [reply lib/hsa/mcwamp_hsa.cpp]: done

File: lib/hsa/mcwamp_hsa.cpp (hunk 2 above)
Comment (david-salinas) [reply lib/hsa/mcwamp_hsa.cpp]: done

**Review by scchan — APPROVED** [review]
LGTM

**Pull request merged**